### DISCUSSION TITLE
Livewire v3 listen to any update events from other components on page to show loading spinner

### DISCUSSION THREAD

**Discussion by akulmehta** [livewire/livewire#8127] [3. Feature ideas]
I have a page with several components and I think rather than adding a wire.loading in each, it would be great if we could add a common wire.loading element to the parent page to show a loading spinner overlay. I have created a loading spinner overlay which listens to dispatched events and show the spinner with class and blade files provided below. 

Is there any way to check for update events from other components on the page to show the same spinner?

Class:

```php
<?php

namespace App\Livewire\Inputs;

use Livewire\Component;

class LoadingOverlay extends Component
{

    protected $listeners = [
        'typeSelected' => 'showLoadingOverlay',
        'loadStepToEdit' => 'showLoadingOverlay',
        'refreshStepToEdit' => 'showLoadingOverlay',
    ];


    public function showLoadingOverlay()
    {
    }

    public function render()
    {
        return view('livewire.inputs.loading-overlay');
    }
}

```

Blade (Bootstrap): 

```blade
<div>
    <div
        class="modal-backdrop show fade bg-body-tertiary"
        wire:loading
    >
        <div class="position-absolute top-50 start-50 translate-middle">
            <div class="spinner-border" style="width: 3rem; height: 3rem;" role="status">
                <span class="visually-hidden">Loading...</span>
            </div>              
        </div>
    </div>
</div>
```

**Comment by joshhanley** [comment]
@akulmehta I don't think there is an inbuilt way at the moment. I have changed this over to a feature request. PRs are welcome! 🙂

**Reply by akulmehta** [reply]
Unfortunately, I am definitely not experienced enough to do a PR. But I hope someone provides a solution.
